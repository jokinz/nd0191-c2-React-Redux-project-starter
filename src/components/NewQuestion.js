import { useEffect, useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions";

function NewQuestion(props) {
  const navigate = useNavigate();

  // console.log("new question props:", props);
  const author = props.activeUser;
  const [optionOneText, setOptionOneText] = useState("");
  const [optionTwoText, setOptionTwoText] = useState("");
  const handleOptionOneChange = (e) => {
    setOptionOneText(e.target.value);
  };
  const handleOptionTwoChange = (e) => {
    setOptionTwoText(e.target.value);
  };

  useEffect(() => {
    setOptionOneText("");
    setOptionTwoText("");
  }, []);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (optionOneText !== "" && optionTwoText !== "") {
      const question = { optionOneText, optionTwoText, author };
      props.dispatch(handleAddQuestion(question)).then(() => {
        navigate("/");
        setOptionOneText("");
        setOptionTwoText("");
      });
    }
  };

  return (
    <div className="new-question-container">
      <h1 className="title">New question </h1>
      <h2 className="new-question-would-you align-center">Would you rather</h2>
      <input
        data-testid="option-one-input"
        className="new-question-input"
        value={optionOneText}
        onChange={handleOptionOneChange}
        placeholder="Option 1"
      ></input>
      <span>or</span>
      <input
        data-testid="option-two-input"
        className="new-question-input"
        value={optionTwoText}
        onChange={handleOptionTwoChange}
        placeholder="Option 2"
      ></input>
      <button
        className="new-question-submit-btn submit-btn"
        onClick={handleSubmit}
      >
        Submit
      </button>
    </div>
  );
}

const mapStateToProps = (state) => {
  return state.activeUser;
};
export default connect(mapStateToProps)(NewQuestion);
