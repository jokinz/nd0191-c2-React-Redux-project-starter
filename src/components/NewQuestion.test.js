import { fireEvent, render } from "@testing-library/react";
import NewQuestion from "./NewQuestion";
import { createStore } from "redux";
import reducer from "../reducers";
import middlewares from "../middlewares";
import "@testing-library/jest-dom";
import { Router } from "react-router-dom";

describe("NewQuestion", () => {
  it("matches the snapshot test", () => {
    const store = createStore(reducer, middlewares);

    var component = render(
      <Router location={"/"}>
        <NewQuestion store={store} activeUser={"sarahedo"} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });
  it("changes the input content for the option 1", () => {
    const store = createStore(reducer, middlewares);

    var component = render(
      <Router location={"/"}>
        <NewQuestion store={store} activeUser={"sarahedo"} />
      </Router>
    );
    var optionOneInput = component.getByTestId("option-one-input");
    const changeOptionInput = () => {
      fireEvent.change(optionOneInput, { target: { value: "new text" } });
    };
    changeOptionInput();
    expect(optionOneInput.value).toBe("new text");
  });
  it("changes the input content for the option 2", () => {
    const store = createStore(reducer, middlewares);

    var component = render(
      <Router location={"/"}>
        <NewQuestion store={store} activeUser={"sarahedo"} />
      </Router>
    );
    var optionTwoInput = component.getByTestId("option-two-input");
    const changeOptionInput = () => {
      fireEvent.change(optionTwoInput, { target: { value: "new text" } });
    };
    changeOptionInput();
    expect(optionTwoInput.value).toBe("new text");
  });
});
