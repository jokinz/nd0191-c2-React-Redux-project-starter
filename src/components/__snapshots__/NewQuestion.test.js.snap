// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion matches the snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="new-question-container"
      >
        <h1
          class="title"
        >
          New question 
        </h1>
        <h2
          class="new-question-would-you align-center"
        >
          Would you rather
        </h2>
        <input
          class="new-question-input"
          data-testid="option-one-input"
          placeholder="Option 1"
          value=""
        />
        <span>
          or
        </span>
        <input
          class="new-question-input"
          data-testid="option-two-input"
          placeholder="Option 2"
          value=""
        />
        <button
          class="new-question-submit-btn submit-btn"
        >
          Submit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="new-question-container"
    >
      <h1
        class="title"
      >
        New question 
      </h1>
      <h2
        class="new-question-would-you align-center"
      >
        Would you rather
      </h2>
      <input
        class="new-question-input"
        data-testid="option-one-input"
        placeholder="Option 1"
        value=""
      />
      <span>
        or
      </span>
      <input
        class="new-question-input"
        data-testid="option-two-input"
        placeholder="Option 2"
        value=""
      />
      <button
        class="new-question-submit-btn submit-btn"
      >
        Submit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
